api/src/main/java/uk/gov/dvsa/mot/app/TradeServiceRequestHandler.java

```diff
@@ -247,7 +247,11 @@ public Vehicle getLatestMotTest(TradeServiceRequest request, Context context) th
                 Vehicle vehicle = tradeReadService.getLatestMotTestByRegistration(registration);
 
                 if (vehicle == null) {
-                    throw new InvalidResourceException("No MOT Test found for registration " + registration, context.getAwsRequestId());
+                    vehicle = tradeReadService.getDvlaVehicleByRegistration(registration);
```

Comment:
This logic should be moved down a level to TradeReadService. The Handlers are instantiated by AWS, which then bootstrap "our" app using Guice and call into our application, using one-to-one calls. Currently, this handler will open and close two database connections because the "transaction" boundary is at the service level and not the Handler level. Not ideal but it's useful to consider the Handlers as an AWS managed endpoint class that calls into our Guice application.